sharedbuffers/mapped_struct.py

```diff
@@ -742,15 +742,15 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0,
                         # inline unsigned ints
                         dtype = b'I'
                         buf[offs] = ord(dtype)
-                    elif (cython.cast(cython.longlong, -0x8000000000000000) <= iminval
-                            and imaxval <= cython.cast(cython.longlong, 0x7FFFFFFFFFFFFFFF)):
+                    elif (cython.cast(cython.longlong, long(-0x8000000000000000)) <= iminval
+                            and imaxval <= cython.cast(cython.longlong, long(0x7FFFFFFFFFFFFFFF))):
```

Comment:
Ok, I see that calling `long` is expensive here. But why did we have the "L" suffix in the first place?  It seems to me it doesn't change anything

```python
cython.cast(cython.longlong, -0x8000000000000000L)
```

turns into
```c
(PY_LONG_LONG)-9223372036854775808L)
```

and 
```python
cython.cast(cython.longlong, -0x8000000000000000)
```
into
```c
(PY_LONG_LONG)-9223372036854775808LL)
```

Reply:
Did you try python 2?

Reply:
TBH, I think the correct version is the second version, because it produces a LL literal, which is the right kind. But I suspect that's not the case in py2 because we've had issues with that in the past, where it would use a regular int literal that gets truncated.

Reply:
py2:
```
+4: x = cython.cast(cython.longlong, -0x8000000000000000)

  __pyx_v_4test_x = ((PY_LONG_LONG)-9223372036854775808LL);
```

py3
```
+4: x = cython.cast(cython.longlong, -0x8000000000000000)

  __pyx_v_4test_x = ((PY_LONG_LONG)-9223372036854775808LL);
```

Reply:
Looks like removing the L suffix is the right way to go about it. Must have been fixed in sme cython patch because it was needed before.

`\_(0_0)_/`